sdk/digitaltwins/digitaltwins/review/digitaltwins.api.md

```diff
@@ -0,0 +1,372 @@
+    eTag: string;
+}
+
+// @public
+export interface DigitalTwinsAddRelationshipOptionalParams extends coreHttp.RequestOptionsBase {
+    ifNoneMatch?: IfNoneMatch1;
+    relationship?: any;
+}
+
+// @public
+export type DigitalTwinsAddRelationshipResponse = DigitalTwinsAddRelationshipHeaders & {
+    body: any;
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: DigitalTwinsAddRelationshipHeaders;
+        bodyAsText: string;
+        parsedBody: any;
+    };
+};
+
+// @public
+export type DigitalTwinsAddResponse = DigitalTwinsAddHeaders & {
+    body: any;
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: DigitalTwinsAddHeaders;
+        bodyAsText: string;
+        parsedBody: any;
+    };
+};
+
+// @public
+export class DigitalTwinsClient {
+    constructor(endpoint: string, credential: TokenCredential | ServiceClientCredentials, options?: AzureDigitalTwinsAPIOptions);
+    createModels(models: any[]): Promise<DigitalTwinModelsAddResponse>;
+    decomissionModel(modelId: string, updateModel: any[]): Promise<RestResponse>;
+    deleteDigitalTwin(digitalTwinId: string, ifMatch?: string): Promise<RestResponse>;
+    deleteEventRoute(eventRouteId: string): Promise<RestResponse>;
+    deleteModel(modelId: string): Promise<RestResponse>;
+    deleteRelationship(digitalTwinId: string, relationshipId: string, ifMatch: string): Promise<RestResponse>;
+    getComponent(digitalTwinId: string, componentPath: string): Promise<DigitalTwinsGetComponentResponse>;
+    getDigitalTwin(digitalTwinId: string): Promise<DigitalTwinsGetByIdResponse>;
+    getEventRoute(eventRouteId: string): Promise<EventRoutesGetByIdResponse>;
+    getModel(modelId: string): Promise<DigitalTwinModelsGetByIdResponse>;
+    getRelationship(digitalTwinId: string, relationshipId: string): Promise<DigitalTwinsGetByIdResponse>;
+    listEventRoutes(maxItemCount?: number): PagedAsyncIterableIterator<EventRoute, EventRoutesListNextResponse>;
+    listIncomingRelationships(digitalTwinId: string, options?: RequestOptionsBase & PageSettings): PagedAsyncIterableIterator<IncomingRelationship, DigitalTwinsListIncomingRelationshipsResponse>;
+    listModels(dependeciesFor?: string[], includeModelDefinition?: boolean, maxItemCount?: number): PagedAsyncIterableIterator<ModelData, DigitalTwinModelsListResponse>;
+    listRelationships(digitalTwinId: string, options?: DigitalTwinsListRelationshipsOptionalParams & PageSettings): PagedAsyncIterableIterator<any>;
+    publishComponentTelemetry(digitalTwinId: string, componentPath: string, payload: string, messageId?: string): Promise<RestResponse>;
+    publishTelemetry(digitalTwinId: string, payload: any, messageId?: string): Promise<RestResponse>;
+    queryTwins(query?: string): PagedAsyncIterableIterator<any, QueryQueryTwinsResponse>;
+    updateComponent(digitalTwinId: string, componentPath: string, componentPatch: any, ifMatch?: string): Promise<DigitalTwinsUpdateComponentResponse>;
```

Comment:
componentPatch can likely be typed more precisely than `any`. Even `object` might be preferable if we can't be more specific.